### ISSUE TITLE
Skip/fast forward doesn't work with downloaded videos on YouTube

### ISSUE THREAD

**Issue by KeraTerra** [matejdro/WearMusicCenter#16]
Steps to recreate: download a video on YouTube app (premium is required). Play a video in the background and block the screen.
On WearMusicCenter, fast forward/backward actions won't work. Play/pause and next track/last track is working. 

Another possible issue is that YouTube app doesn't provide the playlist on WearMusicCenter app. It says "Not supported by this media app". Not sure if it's by design as YouTube surely has playlists feature (e.g. downloaded videos, etc.).

**Comment by matejdro** [comment]
Does this also happen with foreground youtube? I don't have premium so it is a bit hard to reproduce for me.

**Comment by KeraTerra** [comment]
No, it only happens when the screen is locked. When it's open, the YouTube controls fine, even if it's minimized. Also, only happens to downloaded videos. The videos that are streamed via Wi-FI or cellular don't have this problem.

**Comment by matejdro** [comment]
Does skipping works in default music app on your watch?

**Comment by KeraTerra** [comment]
There is no "skipping/fast forward" in the regular Samsung music app on the GW4. Fast forward function is usually only available in podcast apps. For regular video and/or tracks it's only next track/last track.

That's why your app is helpful. It provides fast forward on a global basis, but for some reason it doesn't work with videos downloaded from YouTube for offline playback.

**Comment by matejdro** [comment]
I don't think there is much I can do. What my app does is just send a signal to the app "hey, would you skip next 30 sedoncs please?". And then app decides whether to obey that or not.

It looks like Youtube app does not support skipping in offline mode.

**Comment by KeraTerra** [comment]
Oh, but the Youtube app on the phone does allow skipping in the offline
mode. It happens by double tapping the screen. It's using the same video
player and interface as with regular feed on Youtube.

That's why I find it weird that it works with your app when playing the
videos online, but doesn't work when I launch the downloaded video. It's
just a different playlist on Youtube, that's all.

**Comment by matejdro** [comment]
Can you try seeking via notification when you are playing offline videp?

**Comment by Motisu** [comment]
I installed the app yesterday on my GW4 and it works very nicely, but this bug also exists with other audio apps. I am trying to use it to control Listen Audiobook player on my phone (Xiaomi X3 NFC) and it only partially works. It does start and stop play and it does load the app, but I set up the right touch to single press REV  and double press to REW for 20 sec but it doesn't work. I also set up long press to SKIP PREV but that only rewinds 10 sec. I also set up the hard back button to play Listen audiobook player on 1 press and play on double press but it just ignores it and does a task list or goes into Pay.

**Comment by matejdro** [comment]
Does it work with other audio apps? Then likely this audio app is the culprit and does not receive the commands properly.

Does seeking via seekbar in the notification work?

**Comment by Motisu** [comment]
OK. it seems that you are right about it being an app problem.  I tried it with Pulsar and all controls work as they should. what is interesting is that the Listen controls in the notifications do work as they should. there are some settings in Listen that define remote buttons behavior so I will play around with them and see if that will help.
The hard buttons don't work in either app.

**Comment by matejdro** [comment]
I meant, what happens if you try to use seekbar inside notification?

![Uploading image.png…]()

**Comment by Motisu** [comment]
You are correct again. Seeking in the notification doesn't work either. But the skip forward and rewind do work in the notification.

**Comment by matejdro** [comment]
Yeah, unfortunately, Music Center relies on the notification seeking method. Maybe you can contact the author of the media app to get this fixed?

**Comment by Motisu** [comment]
it would be nice, but its not worth the effort.  slightly less convenient, but I can manage with it the way it is.
All in all, the app is a very useful tool.  Thank you for it!
